test/index.js

```diff
@@ -1356,142 +736,86 @@ describe('request()', () => {
-        });
-
-        server.listen(0, () => {
-
-            Wreck.request('head', 'http://localhost:' + server.address().port, { payload: null }, (err, res) => {
-
-                expect(err).to.not.exist();
-                Wreck.read(res, null, (err, body) => {
+    it('requests head method', async () => {
 
-                    expect(err).to.not.exist();
-                    expect(body.toString()).to.equal('');
-                    server.close();
-                    done();
-                });
-            });
-        });
+        const server = await internals.server('echo');
+        const res = await Wreck.request('head', 'http://localhost:' + server.address().port, { payload: null });
+        const body = await Wreck.read(res);
+        expect(body.toString()).to.equal('');
+        server.close();
     });
 
-    it('post null payload', (done) => {
+    it('post null payload', async () => {
 
-        const server = Http.createServer((req, res) => {
+        const handler = (req, res) => {
 
             res.statusCode = 500;
             res.end();
-        });
-
-        server.listen(0, () => {
-
-            Wreck.request('post', 'http://localhost:' + server.address().port, { headers: { connection: 'close' }, payload: null }, (err, res) => {
-
-                expect(err).to.not.exist();
-                Wreck.read(res, null, (err, body) => {
+        };
 
-                    expect(err).to.not.exist();
-                    expect(body.toString()).to.equal('');
-                    server.close();
-                    done();
-                });
-            });
-        });
+        const server = await internals.server(handler);
+        const res = await Wreck.request('post', 'http://localhost:' + server.address().port, { headers: { connection: 'close' }, payload: null });
+        const body = await Wreck.read(res);
+        expect(body.toString()).to.equal('');
+        server.close();
     });
 
-    it('handles read timeout', (done) => {
+    it('handles read timeout', async () => {
 
-        const server = Http.createServer((req, res) => {
+        const handler = (req, res) => {
 
             setTimeout(() => {
 
                 res.writeHead(200);
                 res.write(internals.payload);
-                res.end();
-            }, 2000);
-        });
-
-        server.listen(0, () => {
-
-            Wreck.request('get', 'http://localhost:' + server.address().port, { timeout: 100 }, (err, res) => {
+                res.end();
+            }, 2000);
+        };
 
-                expect(err).to.exist();
-                expect(err.output.statusCode).to.equal(504);
-                done();
-            });
-        });
+        const server = await internals.server(handler);
+        try {
+            await Wreck.request('get', 'http://localhost:' + server.address().port, { timeout: 100 });
+        }
+        catch (err) {
+            expect(err.output.statusCode).to.equal(504);
+        }
```

Comment:
This is not a safe way to test. The `expect()` won't be called if the request is successful, causing it to unexpectedly pass.

We probably need something like https://github.com/hapijs/code/issues/106, for a simple safe solution.